WordPress/Classes/Extensions/UIViewController+Helpers.swift

```diff
@@ -10,4 +12,16 @@ extension UIViewController
         
         return visibleAsRoot || visibleAsTopOnStack || visibleAsPresented
     }
+    
+    // Determines if the current ViewController is being presented modally
+    //
+    public func isModal() -> Bool {
+        return self.presentingViewController != nil
+    }
+    
+    // Determines if the current ViewController is the only one in the Navigation stack
+    //
+    public func isRootInNavigation() -> Bool {
+        return navigationController?.viewControllers.count == 1
+    }
```

Comment:
@koke my initial thought aligns precisely with what you've outlined. Whoever presents a VC should be the one who actually dismisses it.

Problem arises when we use something designed to be generic, such as `ImmuTableViewController`. Main reason why i ended up going with this design, is that `ImmuTableViewController` doesn't have knowledge of "What VC is being displayed" (and it shouldn't have to), so it was cleaner to add "dismissal intelligence".

Alternative to this is to subclass ImmuTableViewController, directly, and take over UITableView's delegate methods (*OR* overload the ImmuTableVC API, but that sounds a bit like overengineering to me).